dali/kernels/kernel_manager.h

```diff
@@ -218,9 +219,15 @@ class DLL_PUBLIC KernelManager {
    */
   template <typename Kernel, typename... OutInArgs>
   void Run(int thread_idx, int instance_idx, KernelContext &context, OutInArgs &&...out_in_args) {
-    assert(static_cast<size_t>(thread_idx) < scratchpads.size());
-    auto &sa = GetScratchpadAllocator(thread_idx);
-    Run<Kernel>(sa, instance_idx, context, std::forward<OutInArgs>(out_in_args)...);
+    assert(instance_idx >= 0 &&
+           static_cast<size_t>(instance_idx) < NumInstances() &&
+           "Kernel instance index (instance_idx) out of range");
+    auto &inst = instances[instance_idx];
+    DynamicScratchpad scratchpad({}, AccessOrder(context.gpu.stream));
+    auto *old_scratchpad = context.scratchpad;
```

Comment:
Why do we need `context.scratchpad` for if we are not using it anyway here?

Reply:
It's used by the `Run` method invoked afterwards.

Reply:
I was rather asking about the ideas of replacing the old one with one created here.

Reply:
Well, now the old one is always NULL, so we don't need to store it. I'll simplify it (again, possibly in a follow-up).

Reply:
Ok, let do that in a follow-up.

Reply:
So kernel manager `scratchpads` are also to be removed in a follow-up, right?

Reply:
Yes, that's what I brought up in open floor. Since this involves an API change (we no longer need thread_idx), the change affects a large number of files - hence the decision to make the more important change that potentially affects performance (this one) first and then clean the rest.